plexapi/library.py

```diff
@@ -294,8 +294,13 @@ def add(self, name='', type='', agent='', scanner='', location='', language='en'
                   40:South Africa, 41:Spain, 42:Sweden, 43:Switzerland, 44:Taiwan, 45:Trinidad,
                   46:United Kingdom, 47:United States, 48:Uruguay, 49:Venezuela.
         """
-        part = '/library/sections?name=%s&type=%s&agent=%s&scanner=%s&language=%s&location=%s' % (
-            quote_plus(name), type, agent, quote_plus(scanner), language, quote_plus(location))  # noqa E126
+        if isinstance(location, str):
+            if not utils.pathExist(location):
```

Comment:
Using `os.path.isdir(path)` won't work if plexapi is on a different computer than the Plex server.

Use `self._server.browse(path)` instead which mimics Plex Web. I don't know if `server.browse(path)` returns any type of error if the path doesn't exists. See below for checking the path exists.

Only convert the string into a list here.

Reply:
I tried `.browse()` initially but iirc it returns `None` if the path is empty or does not exist.

Reply:
Hmm, I need to think about this.